sys/linux/socket_ip_tunnel.txt

```diff
@@ -0,0 +1,72 @@
+# Copyright 2020 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+include <uapi/linux/if_tunnel.h>
+include <uapi/linux/if.h>
+include <uapi/linux/in.h>
+include <uapi/linux/ip.h>
+include <net/ip.h>
+
+# Fallback tunnels and placeholder names "syztnl*".
+ipv4_tunnel_names = "", "ip_vti0", "tunl0", "gre0", "gretap0", "erspan0", "syztnl0", "syztnl1", "syztnl2"
+ipv6_tunnel_names = "", "ip6tnl0", "ip6_vti0", "ip6gre0", "sit0", "syztnl0", "syztnl1", "syztnl2"
+
+ip_tunnel_flags = GRE_CSUM, GRE_ROUTING, GRE_KEY, GRE_SEQ, GRE_STRICT, GRE_REC, GRE_ACK, GRE_FLAGS, GRE_VERSION, VTI_ISVTI
+ip_tunnel_protocols = 0, IPPROTO_IPIP, IPPROTO_GRE, IPPROTO_IPV6
```

Comment:
From [here](https://github.com/google/syzkaller/pull/1856#discussion_r445848023) I learned that  syzkaller determines whether it is an enum or bitflags by testing if their bits are disjoint and if 0 is among the flags. Because I cherry-picked only some values out of the enum, I added a zero to be sure that they will not be treated as bitflags. They are indeed not disjoint (`IPPROTO_IPIP = 0b000100` and `IPPROTO_GRE = 0b101111`) but that does not seem obvious.
Not sure what is the best practice?

Reply:
Please see #1921

Reply:
Got it. I will remove the 0.